pkg/diagnostics/http_monitoring.go

```diff
@@ -162,7 +186,12 @@ func (h *httpMetrics) ClientRequestStarted(ctx context.Context, method, path str
 		return
 	}
 
-	if h.legacy {
+	matchedPath, ok := h.matchPath(path, Ingress)
+	if ok {
+		path = matchedPath
+	}
+
+	if h.legacy || h.pathMatching.enabled {
```

Comment:
Why does having path matching rules enabled put us in the legacy code path?

Reply:
I changed that after this comment https://github.com/dapr/dapr/pull/7746#discussion_r1608424871

With path matching the keys used are the same as legacy code path the goal here was to simplify the code since these keys are not used in low cardinality.

Reply:
path matching is orthogonal to low/high cardinality so we need to have this in the legacy code path to allow path matching in legacy (high cardinality)